packages/analytics/analytics-utilities/src/timeframes.ts

```diff
@@ -26,9 +28,13 @@ const adjustForTz = (d: Date, tz: string) => {
   return new Date(d.getTime() - getTimezoneOffset(tz, d))
 }
 
+const { i18n } = composables.useI18n()
+
 export class Timeframe {
   readonly timeframeText: string
 
+  readonly rangeDisplayText: string
```

Comment:
One thought...

All timeframes have unique keys.  Instead of adding i18n to `timeframes` (which might cause some issues down the line, since this library is used server-side), maybe the metric card provider can do something like

```
i18n.t(`timeframe.rangeDisplay.${timeframe.key}`)
```

with appropriate adjustments for `custom` timeframes.

Sorry that I didn't think of this earlier when you were talking about it; I hadn't dug into this yet.